pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ThetaSketchIntegrationTest.java

```diff
@@ -0,0 +1,263 @@
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.utils.builder.TableConfigBuilder;
+import org.apache.pinot.util.TestUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+
+public class ThetaSketchIntegrationTest extends BaseClusterIntegrationTest {
+
+  private static final String DIM_NAME = "dimName";
+  private static final String DIM_VALUE = "dimValue";
+  private static final String SHARD_ID = "shardId";
+  private static final String THETA_SKETCH = "thetaSketchCol";
+
+  @BeforeClass
+  public void setup()
+      throws Exception {
+    TestUtils.ensureDirectoriesExistAndEmpty(_tempDir, _segmentDir, _tarDir);
+
+    // Start the Pinot cluster
+    startZk();
+    startController();
+    startBroker();
+    startServer();
+
+    // create & upload schema AND table config
+    Schema schema = new Schema.SchemaBuilder().setSchemaName(DEFAULT_SCHEMA_NAME)
+        .addSingleValueDimension(DIM_NAME, FieldSpec.DataType.STRING)
+        .addSingleValueDimension(DIM_VALUE, FieldSpec.DataType.STRING)
+        .addSingleValueDimension(SHARD_ID, FieldSpec.DataType.INT)
+        .addSingleValueDimension(THETA_SKETCH, FieldSpec.DataType.BYTES).build();
+    addSchema(schema);
+    TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName(DEFAULT_TABLE_NAME).build();
+    addTableConfig(tableConfig);
+
+    // create & upload segments
+    File avroFile = createAvroFile();
+    ClusterIntegrationTestUtils.buildSegmentFromAvro(avroFile, tableConfig, schema, 0, _segmentDir, _tarDir);
+    uploadSegments(DEFAULT_TABLE_NAME, _tarDir);
+
+    waitForAllDocsLoaded(60_000);
+  }
+
+  @Override
+  protected long getCountStarResult() {
+    /*
+    Uploaded table content:
+
+    row#  dimName  dimValue  shardId  thetaSketchCol
+    ----  =======  ========  =======  ==============
+    1     country  US        1        ...
+    2     country  CA        1        ...
+    3     country  MX        1        ...
+    4     title    Engineer  1        ...
+    5     title    Manager   1        ...
+    6     country  US        2        ...
+    7     country  CA        2        ...
+    8     country  MX        2        ...
+    9     title    Engineer  2        ...
+    10    title    Manager   2        ...
+     */
+    return 10;
+  }
+
+  @Test
+  public void testThetaSketchQuery()
+      throws Exception {
+    /*
+    Original data:
+
+    Title     Country  Shard#1  Shard#2
+    --------  -------  -------  -------
+    Engineer  US       50       110
+    Engineer  CA       60       120
+    Engineer  MX       70       130
+    Manager   US       80       140
+    Manager   CA       90       150
+    Manager   MX       100      160
+     */
+
+    // title = engineer
+    String query = "select distinctCountThetaSketch(thetaSketchCol, '', "
```

Comment:
Would be good to create a utility method that generates the query based on parameters. Also, would be good to cover SQL/PQL, selection and group by.

Reply:
I just added the test case for groupby and also ran everything against sql endpoint as well.
For utility method, I intentionally didn't use it here because this is a slightly complicated aggregation function and I wanted to spell out the queries so the reader can easily see what the syntax is and what the differences are.